src/main/java/com/zaxxer/hikari/util/UtilityElf.java

```diff
@@ -165,30 +164,14 @@ public static ThreadPoolExecutor createThreadPoolExecutor(final BlockingQueue<Ru
     * @param transactionIsolationName the name of the transaction isolation level
     * @return the int value of the isolation level or -1
     */
-   public static int getTransactionIsolation(final String transactionIsolationName)
-   {
+   public static int getTransactionIsolation(final String transactionIsolationName) {
       if (transactionIsolationName != null) {
          try {
             // use the english locale to avoid the infamous turkish locale bug
-            final String upperName = transactionIsolationName.toUpperCase(Locale.ENGLISH);
-            if (upperName.startsWith("TRANSACTION_")) {
-               Field field = Connection.class.getField(upperName);
-               return field.getInt(null);
-            }
-            final int level = Integer.parseInt(transactionIsolationName);
-            switch (level) {
-               case Connection.TRANSACTION_READ_UNCOMMITTED:
-               case Connection.TRANSACTION_READ_COMMITTED:
-               case Connection.TRANSACTION_REPEATABLE_READ:
-               case Connection.TRANSACTION_SERIALIZABLE:
-               case Connection.TRANSACTION_NONE:
-               case SQL_SERVER_SNAPSHOT_ISOLATION_LEVEL: // a specific isolation level for SQL server only
-                  return level;
-               default:
-                  throw new IllegalArgumentException();
-             }
-         }
-         catch (Exception e) {
+            final String upperCaseIsolationLevelName = Optional.of(transactionIsolationName.toUpperCase(Locale.ENGLISH))
```

Comment:
Why `Optional` is used here? It's absolutely redundant

Reply:
Right,please check last commit.